starter-code/public/javascripts/financial-data.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+
+// Handler when the DOM is fully loaded
+document.addEventListener("DOMContentLoaded", function () { 
+  var startdatePicker = document.getElementById('startDatePicker');
+  var enddatePicker = document.getElementById('endDatePicker');
+  var currencyPicker = document.getElementById('currencyPicker');
+  // placeholder values
+  enddatePicker.value = moment().subtract(1, 'days').format('YYYY-MM-DD')
+  startdatePicker.value = moment().subtract(1, 'months').format('YYYY-MM-DD');
+  currencyPicker.value = 'USD';
+
+  startdatePicker.addEventListener('change', event => {
+    getFinanceData();
+  });
+
+  enddatePicker.addEventListener('change', event => {
+    getFinanceData();
+  });
+
+  currencyPicker.addEventListener('change', event => {
+    getFinanceData();
+  })
+
+  // for first run - start 
+  getFinanceData();
+  // for first run - end 
+});
+
+function getFinanceData() {
+  let startDate = document.getElementById('startDatePicker').value;
+  let endDate = document.getElementById('endDatePicker').value;
+  let currency = document.getElementById('currencyPicker').value;
+  var currencySign = getCurrencySign(currency); 
+  axios.get(`http://api.coindesk.com/v1/bpi/historical/close.json?start=${startDate}&end=${endDate}&currency=${currency}`)
+    .then(result => {
+      let financeData = result.data.bpi; 
+      printTheChart(financeData, currencySign);
+      displayMaxMin(Object.values(financeData), currencySign);
+    })
+    .catch(err => {
+      console.log(err);
+    })
+};
+
+function printTheChart(stockData, currencySign) {
+  const labels = Object.keys(stockData); //change object to array (keys)
+  const values = Object.values(stockData); //change object to array (values)
+  
+  // delete old canvas and create new on. This fixes the hover issue between old and new data
+  deleteOldChart(); 
```

Comment:
No need to delete the old chart, this results in a flickering chart when updating the params.

When you don't remove the old chart the data will only be updated in the existing chart.

Reply:
This solves another issue :) When I don't delete it, the chart data points will move when you hover one of the data-points, showing the old data again. After googling, they recommended deleting the chart before displaying it again with the new filter